cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -45,6 +54,31 @@ using internal::checked_cast;
 
 namespace engine {
 
+Status WriteParquetData(const std::string& path,
+                        const std::shared_ptr<fs::FileSystem> file_system,
+                        const std::shared_ptr<Table> input) {
+  EXPECT_OK_AND_ASSIGN(auto buffer_writer, file_system->OpenOutputStream(path));
+  PARQUET_THROW_NOT_OK(parquet::arrow::WriteTable(*input, arrow::default_memory_pool(),
```

Comment:
We are not in the `parquet::` namespace so it is not acceptable to throw.  Best to use `ARROW_RETURN_NOT_OK` instead or if you continue to use expect then use `EXPECT_OK`